### PR TITLE
A4A: Sites dashboard - fixing site sort functionality

### PR DESCRIPTION
**Pull request by coder-karen**
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/163

## Proposed Changes

* This PR fixes the sorting functionality on the A4A sites dashboard.
* It does this by coping over the SiteSort component to A4A and renaming it, then conditionally loading it in the Dataviews component if we are in A4A. The reason it has been copied is because the context in A4A is different than that used in Jetpack Manage.

What doesn't work yet:
* Clicking the preview pane resets the sort, so when the sort is desc it resets it to asc (default) first, meaning the preview pane won't open. Issue for that (for sort, search and filters) is here: https://github.com/Automattic/automattic-for-agencies-dev/issues/164

**Update:** With the latest commits, this part is now working

## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

* Click on the site heading in the sites table (on the arrows) to enable sort. Each click should go between enabling sort, then asc and desc.
* Save a URL with sort fields in it, and manually enter it in a new tab. It should work and retain the correct sort direction.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/sites/sites-dashboard/update-sites-dashboard-url.tsx
Comment (cleacos) [inline client/a8c-for-agencies/sections/sites/sites-dashboard/update-sites-dashboard-url.tsx]: I'm reviewing this part, it took my attention

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dataviews/index.tsx
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dataviews/index.tsx:114]: Probably this won't be needed since we are going to move SitesDataViews to A4A.

Comment (cleacos) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dataviews/index.tsx:114]: It will depend on this PR: https://github.com/Automattic/wp-calypso/pull/88725

Comment (cleacos) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dataviews/index.tsx:114]: Issue to track it and update our imports once that PR is merged: [Update our imports to the SitesDataViews component](https://github.com/Automattic/automattic-for-agencies-dev/issues/208)

**Comment by cleacos** [comment]
> What doesn't work yet:

> Clicking the preview pane resets the sort, so when the sort is desc it resets it to asc (default) first, meaning the preview pane won't open. Issue for that (for sort, search and filters) is here: https://github.com/Automattic/automattic-for-agencies-dev/issues/164

After the last commits those issues are fixed and it's working now.

**Review by cleacos — APPROVED** [review]
LGTM!

@coder-karen ?

**Review by coder-karen — COMMENTED** [review]
Looks good to me too! Can't approve as it's my PR. But this is a ✅

**Pull request merged**
